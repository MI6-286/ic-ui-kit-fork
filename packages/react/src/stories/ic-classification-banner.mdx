import {
  Meta,
  Stories,
  Markdown
} from "@storybook/blocks";
import readme from "../../../web-components/src/components/ic-classification-banner/readme.md";
import * as IcClassificationBannerStories from './ic-classification-banner.stories';

<Meta of={IcClassificationBannerStories} />

<Markdown>{readme}</Markdown>

<Stories />
