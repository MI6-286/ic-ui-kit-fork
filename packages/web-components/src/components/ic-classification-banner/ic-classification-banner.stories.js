import { html } from "lit-html";

const defaultArgs = {
  additionalSelectors: "",
  classification: "official",
  country: "uk",
  inline: true,
  upTo: false,
};

export default {
  title: "Web Components/Classification banner",
  component: "ic-classification-banner",
};

export const Classifications = {
  render: () =>
    html`<ic-classification-banner inline="true"></ic-classification-banner>
      <ic-classification-banner
        classification="official"
        inline="true"
      ></ic-classification-banner>
      <ic-classification-banner
        classification="official-sensitive"
        inline="true"
      ></ic-classification-banner>
      <ic-classification-banner
        classification="secret"
        inline="true"
      ></ic-classification-banner>
      <ic-classification-banner
        classification="top-secret"
        inline="true"
      ></ic-classification-banner>`,

  name: "Classifications",
};

export const UpTo = {
  render: () =>
    html`<ic-classification-banner inline="true"></ic-classification-banner>
      <ic-classification-banner
        classification="official"
        inline="true"
        up-to="true"
      ></ic-classification-banner>
      <ic-classification-banner
        classification="official-sensitive"
        inline="true"
        up-to="true"
      ></ic-classification-banner>
      <ic-classification-banner
        classification="secret"
        inline="true"
        up-to="true"
      ></ic-classification-banner>
      <ic-classification-banner
        classification="top-secret"
        inline="true"
        up-to="true"
      ></ic-classification-banner>`,

  name: "Up to",
};

export const BottomOfPage = {
  render: () =>
    html`<ic-classification-banner
      classification="official"
    ></ic-classification-banner>`,

  name: "Bottom of page",
};

export const Country = {
  render: () =>
    html`<ic-classification-banner
        classification="official"
        inline="true"
        country="us"
      ></ic-classification-banner>
      <ic-classification-banner
        classification="official"
        inline="true"
        country=""
      ></ic-classification-banner>`,

  name: "Country",
};

export const AdditionalSelectors = {
  render: () =>
    html`<ic-classification-banner
      classification="official"
      inline="true"
      additional-selectors="ukic"
    />`,

  name: "Additional selectors",
};

export const Playground = {
  render: (args) =>
    html`<ic-classification-banner
      classification=${args.classification}
      inline=${args.inline}
      up-to=${args.upTo}
      country=${args.country}
      additional-selectors=${args.additionalSelectors}
    ></ic-classification-banner>`,

  args: defaultArgs,

  argTypes: {
    classification: {
      options: [
        "default",
        "official",
        "secret",
        "official-sensitive",
        "top-secret",
      ],

      control: {
        type: "select",
      },
    },
  },

  name: "Playground",
};
