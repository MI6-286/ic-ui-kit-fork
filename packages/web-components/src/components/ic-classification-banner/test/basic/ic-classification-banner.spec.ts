import { newSpecPage } from "@stencil/core/testing";
import { ClassificationBanner } from "../../ic-classification-banner";

describe("ic-classification-banner component", () => {
  it("should render with default classification text when no classification set", async () => {
    const page = await newSpecPage({
      components: [ClassificationBanner],
      html: `<ic-classification-banner></ic-classification-banner>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with official classification text", async () => {
    const page = await newSpecPage({
      components: [ClassificationBanner],
      html: `<ic-classification-banner classification="official"></ic-classification-banner>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with official sensitive classification text", async () => {
    const page = await newSpecPage({
      components: [ClassificationBanner],
      html: `<ic-classification-banner classification="official-sensitive"></ic-classification-banner>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with secret classification text", async () => {
    const page = await newSpecPage({
      components: [ClassificationBanner],
      html: `<ic-classification-banner classification="secret"></ic-classification-banner>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with top secret classification text", async () => {
    const page = await newSpecPage({
      components: [ClassificationBanner],
      html: `<ic-classification-banner classification="top-secret"></ic-classification-banner>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with up to before classification", async () => {
    const page = await newSpecPage({
      components: [ClassificationBanner],
      html: `<ic-classification-banner classification="official" up-to=true></ic-classification-banner>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with a different country when supplied", async () => {
    const page = await newSpecPage({
      components: [ClassificationBanner],
      html: `<ic-classification-banner classification="official" country="us"></ic-classification-banner>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it('should render with no country when supplied ""', async () => {
    const page = await newSpecPage({
      components: [ClassificationBanner],
      html: `<ic-classification-banner classification="official" country=""></ic-classification-banner>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render default banner if no props are passed", async () => {
    const page = await newSpecPage({
      components: [ClassificationBanner],
      html: `<ic-classification-banner></ic-classification-banner>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render default banner if props with empty strings are passed", async () => {
    const page = await newSpecPage({
      components: [ClassificationBanner],
      html: `<ic-classification-banner classification="" country="" additionalSelectors=""></ic-classification-banner>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render default banner if props with undefined are passed", async () => {
    const page = await newSpecPage({
      components: [ClassificationBanner],
      html: `<ic-classification-banner classification=${undefined} country=${undefined} additionalSelectors=${undefined}></ic-classification-banner>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with additional selectors after classification when supplied", async () => {
    const page = await newSpecPage({
      components: [ClassificationBanner],
      html: `<ic-classification-banner classification="official" additional-selectors="ukic"></ic-classification-banner>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
