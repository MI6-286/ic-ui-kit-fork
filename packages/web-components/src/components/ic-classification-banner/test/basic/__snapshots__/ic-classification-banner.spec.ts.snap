// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-classification-banner component should render default banner if no props are passed 1`] = `
<ic-classification-banner>
  <template shadowrootmode="open">
    <banner aria-label="Protective marking" class="classification-banner default">
      <ic-typography variant="caption-uppercase">
        protective marking not set
      </ic-typography>
    </banner>
  </template>
</ic-classification-banner>
`;

exports[`ic-classification-banner component should render default banner if props with empty strings are passed 1`] = `
<ic-classification-banner additionalselectors="" classification="" country="">
  <template shadowrootmode="open">
    <banner aria-label="Protective marking" class="classification-banner default">
      <ic-typography variant="caption-uppercase">
        protective marking not set
      </ic-typography>
    </banner>
  </template>
</ic-classification-banner>
`;

exports[`ic-classification-banner component should render default banner if props with undefined are passed 1`] = `
<ic-classification-banner additionalselectors="undefined" classification="undefined" country="undefined">
  <template shadowrootmode="open">
    <banner aria-label="Protective marking" class="classification-banner default">
      <ic-typography variant="caption-uppercase">
        protective marking not set
      </ic-typography>
    </banner>
  </template>
</ic-classification-banner>
`;

exports[`ic-classification-banner component should render with a different country when supplied 1`] = `
<ic-classification-banner classification="official" country="us">
  <template shadowrootmode="open">
    <banner aria-label="Protective marking" class="classification-banner official">
      <span class="offscreen">
        The protective marking of this page is:
      </span>
      <ic-typography variant="caption-uppercase">
        us official
      </ic-typography>
    </banner>
  </template>
</ic-classification-banner>
`;

exports[`ic-classification-banner component should render with additional selectors after classification when supplied 1`] = `
<ic-classification-banner additional-selectors="ukic" classification="official">
  <template shadowrootmode="open">
    <banner aria-label="Protective marking" class="classification-banner official">
      <span class="offscreen">
        The protective marking of this page is:
      </span>
      <ic-typography variant="caption-uppercase">
        uk official ukic
      </ic-typography>
    </banner>
  </template>
</ic-classification-banner>
`;

exports[`ic-classification-banner component should render with default classification text when no classification set 1`] = `
<ic-classification-banner>
  <template shadowrootmode="open">
    <banner aria-label="Protective marking" class="classification-banner default">
      <ic-typography variant="caption-uppercase">
        protective marking not set
      </ic-typography>
    </banner>
  </template>
</ic-classification-banner>
`;

exports[`ic-classification-banner component should render with no country when supplied "" 1`] = `
<ic-classification-banner classification="official" country="">
  <template shadowrootmode="open">
    <banner aria-label="Protective marking" class="classification-banner official">
      <span class="offscreen">
        The protective marking of this page is:
      </span>
      <ic-typography variant="caption-uppercase">
        official
      </ic-typography>
    </banner>
  </template>
</ic-classification-banner>
`;

exports[`ic-classification-banner component should render with official classification text 1`] = `
<ic-classification-banner classification="official">
  <template shadowrootmode="open">
    <banner aria-label="Protective marking" class="classification-banner official">
      <span class="offscreen">
        The protective marking of this page is:
      </span>
      <ic-typography variant="caption-uppercase">
        uk official
      </ic-typography>
    </banner>
  </template>
</ic-classification-banner>
`;

exports[`ic-classification-banner component should render with official sensitive classification text 1`] = `
<ic-classification-banner classification="official-sensitive">
  <template shadowrootmode="open">
    <banner aria-label="Protective marking" class="classification-banner official-sensitive">
      <span class="offscreen">
        The protective marking of this page is:
      </span>
      <ic-typography variant="caption-uppercase">
        uk official sensitive
      </ic-typography>
    </banner>
  </template>
</ic-classification-banner>
`;

exports[`ic-classification-banner component should render with secret classification text 1`] = `
<ic-classification-banner classification="secret">
  <template shadowrootmode="open">
    <banner aria-label="Protective marking" class="classification-banner secret">
      <span class="offscreen">
        The protective marking of this page is:
      </span>
      <ic-typography variant="caption-uppercase">
        uk secret
      </ic-typography>
    </banner>
  </template>
</ic-classification-banner>
`;

exports[`ic-classification-banner component should render with top secret classification text 1`] = `
<ic-classification-banner classification="top-secret">
  <template shadowrootmode="open">
    <banner aria-label="Protective marking" class="classification-banner top-secret">
      <span class="offscreen">
        The protective marking of this page is:
      </span>
      <ic-typography variant="caption-uppercase">
        uk top secret
      </ic-typography>
    </banner>
  </template>
</ic-classification-banner>
`;

exports[`ic-classification-banner component should render with up to before classification 1`] = `
<ic-classification-banner classification="official" up-to="true">
  <template shadowrootmode="open">
    <banner aria-label="Protective marking" class="classification-banner official">
      <span class="offscreen">
        The protective marking of this page is:
      </span>
      <ic-typography variant="caption-uppercase">
        up to uk official
      </ic-typography>
    </banner>
  </template>
</ic-classification-banner>
`;
